### ISSUE TITLE
Conditional styles marked as unused

### ISSUE THREAD

**Issue by tomauty** [Intellicode/eslint-plugin-react-native#6]
Here's a truncated example:

```javascript
const styles = StyleSheet.create({
  checked: {
    borderColor: colors.green,
    backgroundColor: colors.green,
  },
});
```
```javascript
<View style={[payment.status === 'Checked' && styles.checked]} />
```

`Unused style detected: styles.checked`

I would say if a style can be conditionally applied, it should be marked as 'used'. If you agree I can try for a PR.

**Comment by Intellicode** [comment]
Excellent improvement, I can also try to fix it for you, it is a pretty common case I would think. Let me know!

**Comment by tomauty** [comment]
Looking into it now since I might want to contribute some rules later, but feel free to make the change first. Still learning how the codebase works.

**Comment by Intellicode** [comment]
Alright, let me know if you have any other React Native specific rules that you like to add.

**Comment by Intellicode** [comment]
@tomauty Please let me know if the new rule works for you

**Comment by tomauty** [comment]
Looks good. Thank you!

**Comment by satya164** [comment]
Still having this issue in latest release.

**Comment by Intellicode** [comment]
@satya164 Can you provide an example snippet that triggers the issue for you? It could be an edge case I didn't think of or didn't implement (For example, I don't follow references)

**Comment by Intellicode** [comment]
@satya164 closing the conversation for now, please open a ticket with details if you found a case that's not working for you
